konduit-serving-models/konduit-serving-tensorflow-python/src/main/java/ai/konduit/serving/models/tensorflowpython/KerasStep.java

```diff
@@ -0,0 +1,131 @@
+            return new Runner((KerasStep) step);
+        }
+    }
+
+    @CanRun(KerasStep.class)
+    public static class Runner implements PipelineStepRunner {
+        private final KerasStep step;
+        private final KerasModel model;
+
+        private void validateStep() {
+            if (step.inputKeys == null || step.inputKeys.length == 0) {
+                if (model.numInputs() > 1) {
+                    throw new IllegalArgumentException("Error in KerasStep: Keras model has multiple inputs, but input keys were not provided.");
+                }
+            } else if (step.inputKeys.length != model.numInputs()) {
+                throw new IllegalArgumentException("Error in KerasStep: Keras model has " + model.numInputs() + " inputs but " + step.inputKeys.length + " input keys were provided.");
+            }
+            if (step.outputKeys == null || step.outputKeys.length == 0) {
+                throw new IllegalArgumentException("Error in KerasStep: Output keys not specified");
+            } else if (step.outputKeys.length != model.numOutputs()) {
+                throw new IllegalArgumentException("Error in KerasStep: Keras model has " + model.numOutputs() + " outputs but " + step.outputKeys.length + " output keys were provided.");
+            }
+        }
+
+        public Runner(KerasStep step) {
+            try (PythonGIL gil = PythonGIL.lock()) {
+                this.step = step;
+                this.model = new KerasModel(step.modelUri);
+                validateStep();
+            }
+        }
+
+        @Override
+        public Data exec(Context ctx, Data input) {
+            try (PythonGIL gil = PythonGIL.lock()) {
+                NumpyArray[] inputArrays;
+                if (step.inputKeys == null || step.inputKeys.length == 0) {
+                    String errMultipleKeys = "Multiple NDarray values received for single input model. Specify input key explicitly.";
+                    String errNoKeys = "No NDarray values received.";
+                    String key = DataUtils.inferField(input, ValueType.NDARRAY, false, errMultipleKeys, errNoKeys);
+                    inputArrays = new NumpyArray[]{input.getNDArray(key).getAs(NumpyArray.class)};
+
+                } else {
+                    inputArrays = new NumpyArray[step.inputKeys.length];
+                    for (int i = 0; i < inputArrays.length; i++) {
+                        inputArrays[i] = input.getNDArray(step.inputKeys[i]).getAs(NumpyArray.class);
+                    }
+                }
+                NumpyArray[] out = model.predict(inputArrays);
+                for (int i = 0; i < step.outputKeys.length; i++) {
+                    input.put(step.outputKeys[i], new NumpyArray.NumpyNDArray(out[i]));
+                }
+                return input;
+            }
+        }
+
+
+        @Override
+
+        public void close() {
+
+        }
```

Comment:
If there's a way to close the KerasModel (i.e., deallocate the python object, or at least allow it to be GC'd) we should do it here